### DISCUSSION TITLE
Difference in .env file environment variables injected via docker run vs docker compose

### DISCUSSION THREAD

**Discussion by jimAtLoyal** [containerd/nerdctl#1446] [Q&A] [answered]
### Description

Looks like `docker run` puts single quotes around variables in .env regardless. `docker compose` does not, which is more like the Docker .env file. spec https://docs.docker.com/compose/env-file/ 

### Steps to reproduce the issue

1. Create .env file with the following content
```
ZZ='.env file $test pound # and more'
ZZ_NOQUOTES=.env file $test pound # and more
```

2. Create docker-compose.yml 
```yaml
services:
  busybox:
    image: busybox
    env_file:
      - .env
    command: ['sh', '-c', 'set']
```
3. Run `nerdctl compose up` and see this output:
```
ZZ='.env file $test pound # and more'
ZZ_NOQUOTES='.env file  pound'
```
4. Run `nerdctl run --rm --env-file ./.env  busybox /bin/sh -c set` and see this output
```
ZZ=''"'"'.env file $test pound # and more'"'"
ZZ_NOQUOTES='.env file $test pound # and more'
```

### Describe the results you received and expected

Output from both should be the same

Shouldn't it think $test in the second string is a variable?

### What version of nerdctl are you using?

nerdctl version 0.22.2

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

Rancher Desktop for macOS

### Host information

nerdctl info
Client:
 Namespace:	default
 Debug Mode:	false

Server:
 Server Version: v1.6.8
 Storage Driver: overlayfs
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Log: fluentd journald json-file
  Storage: native overlayfs
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.15.64-0-virt
 Operating System: Alpine Linux v3.16
 OSType: linux
 Architecture: aarch64
 CPUs: 2
 Total Memory: 4.803GiB
 Name: lima-rancher-desktop
 ID: 71ecab30-8f3f-4b45-b916-151c3f9f1d10

**Comment by Zheaoli** [comment, marked as answer]
I think it's not a bug? I guess.

The nerdctl has the same behavior as Docker(`docker run` and `docker-compose up` have different env file inject behavior

**Reply by jimAtLoyal** [reply]
You're right, I got access to a machine w/Docker and `docker run` and `docker compose` behave same as nerdctl (which doesn't match their doc, fwiw) So if you have an .env file, it can't be used for both.
